### PR TITLE
Clarify documentation about customMapStyles being only effective with Google provider

### PR DESCRIPTION
**Pull request by gkpo**
### Does any other open PR do the same thing?

No

### What issue is this PR fixing?

[customMapStyle has no effect when using Apple maps as provider #4948](https://github.com/react-native-maps/react-native-maps/issues/4948)

### How did you test this PR?

This is a change in the Readme.

### CODE REVIEW

**Comment by salah-ghanim** [comment]
@gkpo would be nice if you update the documentations instead in two ways:
1. the section title: Customizing the map style could have (Provider: GoogleMaps) as part of the title
2. mention that this is a depricated way of customyzing style instead using google maps cloud and using mapId is the way to go (this feature was introduced recently but not well documented and doesn't work yet for iOS).

File: README.md [hunk 1]
```diff
@@ -246,7 +246,7 @@ render() {
 }
 ```
 
-### Customizing the map style
+### Customizing the map style (Google Maps provider only)
```
Comment (jan-kozinski) [inline README.md:249]:
```suggestion
### Customizing the map style (Google Maps only)
```

File: README.md [hunk 2]
```diff
@@ -265,7 +265,7 @@ render() {
 }
 ```
 
-For iOS, in addition to providing the `mapStyle` you will need to do the following
+This option only works with the Google provier. So for iOS, in addition to providing the `mapStyle` you will need to do specify the provier with the following:
```
Comment (jan-kozinski) [inline README.md:268]:
```suggestion
For iOS, in addition to providing the `mapStyle` you will need to set the provider to Google Provider as in:
```

Comment (charneykaye) [inline README.md:268]: typo: should be "Google provider"

**Pull request closed without merging**